### PR TITLE
[eclipse/xtext#1679] converted Xtend code to Java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679] converted Xtend code to Java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		return new TypeReference("javax.annotation.Generated");
+	}
+
+	@Override
+	public String toString() {
+		return generate().toString();
+	}
+
+	@Override
+	public boolean equals(final Object obj) {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java:72]: final

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AntlrGrammar.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return getName().replace('.', '/') + ".g";
+	}
+
+	public String getTokensFileName() {
+		return getName().replace('.', '/') + ".tokens";
+	}
+
+	@Override
+	public String toString() {
+		return this.getName();
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AntlrGrammar.java:30]: this.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+	private String comment;
+
+	@Override
+	public CharSequence generate() {
+		StringBuilder stringBuilder = new StringBuilder("@Generated(");
+		if (includeDate || !Strings.isEmpty(comment)) {
+			stringBuilder.append("value = ");
+		}
+		stringBuilder.append("\"");
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java:33]: `append(char)` would relieve you from the escape sequence.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	@Override
+	public String toString() {
+		return generate().toString();
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		return obj instanceof GeneratedClassAnnotation;
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/GeneratedClassAnnotation.java:73]: Unrelated: Subclasses will have different hashCodes but answer true from equals.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/SuppressWarningsAnnotation.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+ * A class annotation configuration for the <code>@SuppressWarnings</code>
+ * annotation.
+ */
+public class SuppressWarningsAnnotation implements IClassAnnotation {
+	private String suppress = "all";
+
+	@Override
+	public CharSequence generate() {
+		StringBuilder stringBuilder = new StringBuilder("@SuppressWarnings(");
+		String[] suppressedWarnings = suppress.split("\\s*,\\s*");
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/SuppressWarningsAnnotation.java:27]: `Splitter.on(',').trimResults()`

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/SuppressWarningsAnnotation.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+		if (suppressedWarnings.length != 1) {
+			stringBuilder.append("{");
+		}
+		String value = Arrays.stream(suppressedWarnings).map((s)-> "\"" + s + "\"").collect(Collectors.joining( ", " ));
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/annotations/SuppressWarningsAnnotation.java:31]: Paranoid:` '"' + Strings.convertToJavaString(s) '"'`

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/util/GrammarUtil2.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.GrammarUtil;
+
+import com.google.common.base.Objects;
+
+public class GrammarUtil2 extends GrammarUtil {
+	public static final String TERMINALS = "org.eclipse.xtext.common.Terminals";
+
+	public static boolean inherits(Grammar grammar, String languageID) {
+		if (Objects.equal(grammar.getName(), languageID)) {
```
Comment (nbhusare) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/util/GrammarUtil2.java:20]: `Objects.equal(grammar.getName(), languageID)` -> `grammar.getName().equals(languageID)`

**Pull request merged**